File: find-smallest-letter-greater-than-target/find-smallest-letter-greater-than-target.py
class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
        l, r = 0, len(letters)-1
        while l <= r:
            m = (l+r)//2
            if letters[m] <= target: l = m+1
            else: r = m-1
        if l == len(letters): return letters[0]
        return letters[l]

        # // Find the letter grather than target

File: find-smallest-letter-greater-than-target/find-smallest-letter-greater-than-target.cpp
class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        int l = 0, r = letters.size()-1, m;
        while(l <= r){
            m = (l+r)/2;
            if(letters[m] <= target) l = m+1;
            else r = m-1;
        }
        return l == letters.size() ? letters[0] : letters[l];
    }
};

// Find smallest letter grather than target